### ISSUE TITLE
Don't download ways with highway=no or highway=razed

### ISSUE THREAD

**Issue by nikita03565** [gboeing/osmnx#937]
- [x] I have read the OSMnx [contributing guidelines](../blob/main/CONTRIBUTING.md)
  - [x] I have read the OSMnx [documentation](https://osmnx.readthedocs.io/) and [usage examples](https://github.com/gboeing/osmnx-examples)
  - [x] I have searched the open/closed issues and pull requests for others similar to mine

**Is your feature proposal related to a problem?**

OSMnx downloads ways with highway types "no" and "razed".
Examples: 
- https://www.openstreetmap.org/way/99396399
- https://www.openstreetmap.org/way/588056038

**Describe the solution you'd like to propose**

I'm thinking to add "no" and "razed" into the filter like so:
![image](https://user-images.githubusercontent.com/33401112/224080896-530e595e-945a-4d41-af2a-2bc4f90105f7.png)

**Describe alternatives you've considered**

Can't think of any alternatives. Perhaps other filters should be updated in same manner.

**Additional context**

```python3
import osmnx as ox
from osmnx import settings
from shapely.geometry.polygon import Polygon
from collections import Counter

settings.use_cache = True
settings.log_file = True

coords = [(59.77605,30.60302),(59.77471,30.68919),(59.75529,30.68190), (59.75577,30.58087)]
boundaries = Polygon([reversed(x) for x in coords])
G = ox.graph_from_polygon(boundaries, network_type="drive")
fig, ax = ox.plot_graph(G)

highway_types = Counter(e[2]["highway"] for e in G.edges(data=True))
print(highway_types)
# Output:
# Counter({'secondary': 19,
#         'tertiary': 23,
#         'no': 2,
#         'unclassified': 24,
#         'residential': 46})

for e in G.edges(data=True):
    if e[2]["highway"] == "no":
        print(e)
# Output:
# (245966921, 1176834042, {'osmid': 99396399, 'highway': 'no', 'oneway': False, 'reversed': False, 'length': 995.6320000000002, 'geometry': <LINESTRING (30.622 59.769, 30.623 59.769, 30.624 59.769, 30.625 59.769, 30....>})
# (1176834042, 245966921, {'osmid': 99396399, 'highway': 'no', 'oneway': False, 'reversed': True, 'length': 995.6320000000002, 'geometry': <LINESTRING (30.637 59.765, 30.636 59.765, 30.636 59.765, 30.636 59.765, 30....>})
```

**Comment by gboeing** [comment]
Thanks. This could be a good suggestion.

Can you link to some example OSM ways that use `highway=no` or `highway=razed`? I know there's a "razed" tag on OSM, but I can't find any documentation for using "razed" as a value with the "highway" tag.

**Comment by nikita03565** [comment]
Thanks for your response!
I already included links in the issue's description:
> Examples:
> - https://www.openstreetmap.org/way/99396399
> - https://www.openstreetmap.org/way/588056038

There are pages in the wiki as well: 
https://wiki.openstreetmap.org/wiki/Tag:highway=no?uselang=en-GB
https://wiki.openstreetmap.org/wiki/Tag:highway=razed?uselang=en-GB

And taginfo:
https://taginfo.openstreetmap.org/tags/highway=no
https://taginfo.openstreetmap.org/tags/highway=razed

**Comment by gboeing** [comment]
Thanks for that. Sorry I had skimmed your issue too quickly.

Ah I see now, so `highway=razed` is deprecated, but it'll still be good to handle the preexisting cases.

**Comment by gboeing** [comment]
closed by #938
